### PR TITLE
feat(eslint-config-fluid): Enable import/no-deprecated

### PR DESCRIPTION
**Pull request by tylerbutler**
Enables the [import/no-deprecated](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-deprecated.md) lint rule in all our configs. I also bumped the config version to a new minor version since 2.0.0 was already released.

I disabled it for test files, since they often need to import deprecated stuff for test purposes.

This rule is useful as we refactor and move APIs to different packages, because it helps us find places where we're using APIs that have been replaced. This will be particularly useful with common-definitions and common-utils.

In my testing, the following packages had legitimate failures with this rule:

- @fluid-internal/test-version-utils
- @fluid-example/webflow
- @fluid-example/table-document
- @fluid-experimental/sequence-deprecated
- @fluid-experimental/attributable-map
- @fluidframework/map
- @fluidframework/merge-tree
- @fluidframework/container-runtime

However, the offenses can be addressed easily or ignored once this change has made it into the client release group.

### CODE REVIEW

File: common/build/eslint-config-fluid/minimal.js [hunk 1]
```diff
@@ -374,6 +375,7 @@ module.exports = {
 				"@typescript-eslint/no-invalid-this": "off",
 				"@typescript-eslint/unbound-method": "off", // This rule has false positives in many of our test projects.
 				"import/no-nodejs-modules": "off", // Node libraries are OK for test files.
+        "import/no-deprecated": "off", // Deprecated APIs are OK to use in test files.
```
Comment (Josmithr) [inline common/build/eslint-config-fluid/minimal.js:378]: Do we want deprecated stuff in our tests? I would argue for having the same rules between test and non-test code as much as possible.

(Also indentation is off on this line)

Comment (Josmithr) [inline common/build/eslint-config-fluid/minimal.js:378]: Indentation is off on this line

**Review by Josmithr — APPROVED** [review]
Only issue is indentation on one of the lines of the config. Otherwise 🚀

File: common/build/eslint-config-fluid/minimal.js (hunk 1 above)
Comment (tylerbutler) [reply common/build/eslint-config-fluid/minimal.js]: Good catch. I also updated the build step to run prettier so this sort of thing shouldn't sneak by in the future.

**Pull request merged**